lib/src/main/java/com/dynatrace/metric/util/DynatraceMetadataEnricher.java

```diff
@@ -39,48 +36,34 @@
    *     with empty key or value are discarded.
    */
   static List<Dimension> getDynatraceMetadata() {
-    return parseDynatraceMetadata(
-        getMetadataFileContentWithRedirection(
-            INDIRECTION_FILE_NAME, ALTERNATIVE_METADATA_FILENAME));
+    return createDimensionList(
+        getPropertiesWithIndirection(INDIRECTION_FILE_NAME, ALTERNATIVE_METADATA_FILENAME));
   }
 
   /**
-   * This function takes a list of strings from the Dynatrace metadata file and transforms it into a
-   * list of {@link Dimension} objects. Parsing failures will not be added to the output list.
-   * Therefore, it is possible that the output list is shorter than the input, or even empty.
+   * This function takes {@link Properties} object and transforms it into a {@link List<Dimension>}.
    *
-   * @param lines a {@link List<String>} containing key-value pairs (as one string) separated by an
-   *     equal sign.
+   * @param properties {@link Properties} to transform
    * @return A {@link List} of {@link Dimension dimensions} mapping {@link String} to {@link
-   *     String}. These represent the the lines passed in separated by the first occurring equal
-   *     sign on each line, respectively. If no line is parsable, returns an empty list. Dimensions
-   *     are not normalized.
+   *     String}. These represent the property entries, where empty keys or values were omitted.
+   *     Dimensions are not normalized.
    */
-  static List<Dimension> parseDynatraceMetadata(Collection<String> lines) {
+  static List<Dimension> createDimensionList(Properties properties) {
     ArrayList<Dimension> entries = new ArrayList<>();
```

Comment:
We could create the list based on the side of the properties.

Reply:
great idea! done in 5f577a0 :)